_source/_posts/2021-11-08-angular-v13.md

```diff
@@ -0,0 +1,183 @@
+---
+layout: blog_post
+title: "What you need to know about Angular v13"
+author: alisa-duncan
+by: advocate
+communities: [security,javascript]
+description: "Highlights of a few Angular v13 features with code snippets"
+tags: [javascript, typescript, angular]
+tweets:
+- "Excited for @angular v13? Check out some of the new features with code demos! #angular"
+- "Check out some use cases for @angular v13 features. What's your favorite feature? #angular"
+- ""
+image:
+type: awareness
+---
+
+Angular v13 arrived! And with it comes a lot of exciting new features and updates. Angular continues to improve run-time performance, decrease compilation time, promote good software development practices, enhance developer experience, and keep up to date with dependencies such as [TypeScript](https://devblogs.microsoft.com/typescript/announcing-typescript-4-4/) and [RxJs](https://rxjs.dev/6-to-7-change-summary). Is anyone else excited about RxJS v7?! 🤩 Let's take a look at a few of the many new exciting features in Angular v13. 
+
+## So long, View Engine!
+
+With v13, View Engine is no longer available and replaced by Ivy. Ivy is the future of Angular applications, and the benefits are tremendous. After initially landing in v12, Ivy paves the way for new feature updates, simplifies existing capabilities, and increases rendering performance.
+
+An example of simplifying existing capabilities is how to create components dynamically. Dynamic components are components loaded at runtime. You might use this when you need to tailor visible components based on responses from a server call. Another example might be if your identity token includes a custom claim, such as department, and you want to show a custom dynamic component specific to departmental information. 
+
+Previously, to dynamically create components, you had to use the `ComponentFactoryResolver`. Now the API takes the Component type directly. Let's look at an example.
+
+In my example, I added a custom claim using the Okta dashboard named `pet_preference`. The app uses the claim value to rotate through images of pets that match their preference.
```

Comment:
context missing, may be a simpler example without involving Okta, else may be a line or two setting the context otherwise someone could be confused why we are talking about claims and stuff

Reply:
If you wanted to make this into a conversion post, you could do it in a couple sentences with the Okta CLI. It's as simple as installing the CLI and running `okta start angular`. https://developer.okta.com/blog/2020/12/03/angular-okta

Reply:
Yes, definitely an option. Would we want more step-by-step instructions to do so? There's missing code for authenticating, calling the service, things like that.

Reply:
I will turn the sample app that I created for this into a separate conversion post.